python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -1143,6 +1143,11 @@ def make_asset_deps(deps: Optional[Iterable[CoercibleToAssetDep]]) -> Optional[I
     if deps is None:
         return None
 
+    # when AssetKey was a plain NamedTuple, it also happened to be Iterable[CoercibleToAssetKey]
+    # so continue to support it here
+    if isinstance(deps, AssetKey):
+        deps = [deps]
```

Comment:
Can you put in a warning here that we will no longer accept this soon.

Also is the "correct" (backwards compat) behavior not to generate N deps, one for each asset key part?

Reply:
iterating over the key tuple was equivalent to `(path, )` or `Tuple[List[str]]` so you just get the one `path` list and coerce that back in to the same `AssetKey`